engine/src/main/java/org/camunda/bpm/engine/runtime/VariableInstanceQuery.java

```diff
@@ -64,6 +64,16 @@
   /** Only select variable instances which have one of the activity instance ids. **/
   VariableInstanceQuery activityInstanceIdIn(String... activityInstanceIds);
 
+  /**
+   * The query will match the names of variables in a case-insensitive way.
+   */
+  VariableInstanceQuery matchVariableNamesIgnoreCase();
+
+  /**
+   * The query will match the values of variables in a case-insensitive way.
+   */
+  VariableInstanceQuery matchVariableValuesIgnoreCase();
```

Comment:
Explicitly document that it pairs only with `variableValueXXX`. (it won't work with `variableNameXXX`)